test/parallel/test-worker-init-failure.js

```diff
@@ -30,12 +30,14 @@ if (process.argv[2] === 'child') {
     });
 
     // We want to test that if there is an error in a constrained running
-    // environment, it will be one of `EMFILE` or `ERR_WORKER_INIT_FAILED`.
+    // environment, it will be one of `ENFILE`, `EMFILE`, 'ENOENT', or
+    // `ERR_WORKER_INIT_FAILED`.
+    const expected = ['ERR_WORKER_INIT_FAILED', 'EMFILE', 'ENFILE', 'ENOENT'];
```

Comment:
When/why is ENOENT returned? I read through the linked PR and I saw mention of ENOENT but no explanation.

Reply:
> When/why is ENOENT returned? I read through the linked PR and I saw mention of ENOENT but no explanation.

If it helps, here's an example of the test failing with ENOENT: https://ci.nodejs.org/job/node-test-commit-osx/35762/nodes=osx1015/console

```
20:31:24 not ok 2477 parallel/test-worker-init-failure
20:31:24   ---
20:31:24   duration_ms: 3.236
20:31:24   severity: fail
20:31:24   exitcode: 1
20:31:24   stack: |-
20:31:24     child stdout: 
20:31:24     
20:31:24     child stderr: events.js:291
20:31:24           throw er; // Unhandled 'error' event
20:31:24           ^
20:31:24     
20:31:24     AssertionError [ERR_ASSERTION]: ENOENT not expected
20:31:24         at Worker.<anonymous> (/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1015/test/parallel/test-worker-init-failure.js:40:14)
20:31:24         at Worker.emit (events.js:314:20)
20:31:24         at Worker.[kOnErrorMessage] (internal/worker.js:241:10)
20:31:24         at Worker.[kOnMessage] (internal/worker.js:251:37)
20:31:24         at MessagePort.<anonymous> (internal/worker.js:167:57)
20:31:24         at MessagePort.[nodejs.internal.kHybridDispatch] (internal/event_target.js:353:41)
20:31:24         at MessagePort.exports.emitMessage (internal/per_context/messageport.js:18:26)
20:31:24         at Worker.[kOnExit] (internal/worker.js:223:5)
20:31:24         at Worker.<computed>.onexit (internal/worker.js:164:20)
20:31:24     Emitted 'error' event on process instance at:
20:31:24         at emitUnhandledRejectionOrErr (internal/event_target.js:542:11)
20:31:24         at MessagePort.[nodejs.internal.kHybridDispatch] (internal/event_target.js:357:9)
20:31:24         at MessagePort.exports.emitMessage (internal/per_context/messageport.js:18:26)
20:31:24         at Worker.[kOnExit] (internal/worker.js:223:5)
20:31:24         at Worker.<computed>.onexit (internal/worker.js:164:20) {
20:31:24       generatedMessage: false,
20:31:24       code: 'ERR_ASSERTION',
20:31:24       actual: false,
20:31:24       expected: true,
20:31:24       operator: '=='
20:31:24     }
20:31:24     
20:31:24     
20:31:24     assert.js:103
20:31:24       throw new AssertionError(obj);
20:31:24       ^
20:31:24     
20:31:24     AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
20:31:24     
20:31:24     1 !== 0
20:31:24     
20:31:24         at ChildProcess.<anonymous> (/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1015/test/parallel/test-worker-init-failure.js:65:12)
20:31:24         at ChildProcess.<anonymous> (/Users/iojs/build/workspace/node-test-commit-osx/nodes/osx1015/test/common/index.js:365:15)
20:31:24         at ChildProcess.emit (events.js:314:20)
20:31:24         at Process.ChildProcess._handle.onexit (internal/child_process.js:276:12) {
20:31:24       generatedMessage: true,
20:31:24       code: 'ERR_ASSERTION',
20:31:24       actual: 1,
20:31:24       expected: 0,
20:31:24       operator: 'strictEqual'
20:31:24     }
20:31:24   ...
```

Reply:
And here's the comment that I believe initially suggested accommodating `ENOENT`: https://github.com/nodejs/node/issues/33759#issuecomment-659816280

@gireeshpunathil

Reply:
I'm able to get ENOENT locally with `tools/test.py -j 16 --repeat 192 test/parallel/test-worker-init-failure.js` on macOS. Happy to take steps to provide more information for debugging purposes.

Reply:
and this is the more low level stack and explanation for `ENOENT` : https://github.com/nodejs/node/issues/33759#issuecomment-663979111

Reply:
Ah, that makes sense. I assumed `getcwd()` would be its own system call (like it is on Linux) but I could've guessed it would've been implemented in terms of `fcntl(F_GETPATH)` on macOS.